feature-wallet-impl/src/main/java/jp/co/soramitsu/wallet/impl/presentation/balance/networkissues/unavailable/NetworkUnavailableViewModel.kt

```diff
@@ -0,0 +1,42 @@
+package jp.co.soramitsu.wallet.impl.presentation.balance.networkissues.unavailable
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import jp.co.soramitsu.account.impl.presentation.AccountRouter
+import jp.co.soramitsu.common.base.BaseViewModel
+import jp.co.soramitsu.wallet.impl.presentation.balance.networkissues.unavailable.NetworkUnavailableFragment.Companion.KEY_PAYLOAD
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.stateIn
+
+@HiltViewModel
+class NetworkUnavailableViewModel @Inject constructor(
+    val savedStateHandle: SavedStateHandle,
+    private val accountRouter: AccountRouter
+) : BaseViewModel() {
+
+    private val chainName: String? = savedStateHandle[KEY_PAYLOAD]
+
+    val state: StateFlow<NetworkUnavailableViewState> = flowOf(
```

Comment:
I think it's not necessary to have a view model here, but if you already have created it - let's keep it.
Same to StateFlow, we can just make the state as a simple string field. Since it is already here - it's up to you.